manifests/server/config_entry.pp

```diff
@@ -98,10 +98,13 @@
   # is managed for us in postgresql::server::config.
   if $facts['os']['name'] == 'Debian' or $facts['os']['name'] == 'Ubuntu' {
     if $name == 'data_directory' {
+      $stop_command = ['service', shell_escape($postgresql::server::service_name), 'stop']
+      $stop_onlyif = ['service', shell_escape($postgresql::server::service_name), 'status']
+      $stop_unless = ["grep \"data_directory = '${shell_escape($value)}'\" ${shell_escape($postgresql::server::postgresql_conf_path)}"]
```

Comment:
Why isn't this a proper array?
```suggestion
      $stop_unless = ['grep', "data_directory = '${shell_escape($value)}'", shell_escape($postgresql::server::postgresql_conf_path)]
```

Reply:
Separating into an array does not seem to work correctly when used with grep

Reply:
> command

As to not needing to shell escape, yeh I was a little over enthusiastic

Reply:
Also hit a similar issue with exec command in passwd.pp

Reply:
What kind of issue? Could it actually be double escaping?

In normal programming (Ruby, Python, C, etc) if you pass an array it gets passed straight into the program so no escaping is needed.

Reply:
Following up, you suggested fix throws this error:
```
Error: /Stage[main]/Postgresql::Server::Config/Postgresql::Server::Config_entry[data_directory]/Exec[postgresql_stop_data_directory]: Could not evaluate: Could not find command 'data_directory'
```

Reply:
Adding the escaped characters thows the same error

Reply:
I think that's because `unless` doesn't implement it in the same way: https://puppet.com/docs/puppet/7/type.html#exec-attribute-unless
You need to pass it as an array within an array. So
```suggestion
      $stop_unless = [['grep', "data_directory = '${value}'", $postgresql::server::postgresql_conf_path]],
```

Reply:
Ok yeh, double wrapping looks to have worked